packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -89,41 +146,80 @@ const getProps = ({
   size,
   theme,
   variant,
+  intent,
+  contrast,
 }: {
   buttonTypographyTokens: ButtonTypography[TypographyPlatforms];
   children?: string;
   isDisabled: boolean;
   theme: Theme;
-  size: Required<BaseButtonCommonProps['size']>;
-  variant: Required<BaseButtonCommonProps['variant']>;
+  size: Required<BaseButtonProps['size']>;
+  variant: Required<BaseButtonProps['variant']>;
+  intent: BaseButtonProps['intent'];
+  contrast: Required<BaseButtonProps['contrast']>;
 }): BaseButtonStyleProps => {
   const props: BaseButtonStyleProps = {
     iconSize: buttonIconSize[size],
     fontSize: buttonTypographyTokens.fonts.size[size],
     lineHeight: buttonTypographyTokens.lineHeights[size],
     minHeight: makeSize(buttonMinHeight[size]),
     iconSpacing: makeSpace(theme.spacing[buttonIconSpacing[size]]),
-    iconColor: `action.icon.${variant}.default`,
-    textColor: `action.text.${variant}.default`,
+    iconColor: getColorToken({
+      property: 'icon',
+      variant,
+      contrast,
+      intent,
+      state: 'default',
+    }) as IconProps['color'],
```

Comment:
Because the `getColorToken` function's return type contains a union of color tokens accepted for icon, text, background, etc. but since here we are enforced to only pass icon color tokens. We'll have to do some generics magic to do this, the use case was too tiny to put in more efforts here but we can look into it.